### PR TITLE
Signup invite code styling

### PR DESCRIPTION
**Pull request by cjb**
@keybase/react-hackers @keybase/design 

This PR also adds a form of &lt;Input/&gt; that we didn't have before -- there is hintText, but no floatingText when you focus, and the hintText should be centered and black10.

Zeplin:

screen shot 2016-03-11 at 1 04 53 pm

GIF:

invite

### CODE REVIEW

**Comment by cecileboucheron** [comment]
Looks great! 
One tiny thing: the period is missing at the end of the error message. Other than that, all good for me.

**Comment by cjb** [comment]
@cecileboucheron Thanks, while I'm changing it maybe we should change the text -- I'll go with:

"Sorry, that's not a valid invite code."

unless someone has a better suggestion soon

File: shared/styles/style-guide.desktop.js [hunk 1]
```diff
@@ -44,7 +44,8 @@ export const globalColors = {
 }
 
 export const globalResizing = {
-  login: {width: 700, height: 580},
+  // Add 25px to login for the OS menubar
+  login: {width: 700, height: 605},
```
Comment (chrisnojima) [inline shared/styles/style-guide.desktop.js]: Is this required. I think the content area matches zeplin as is

File: shared/common-adapters/icon.js.flow [hunk 2]
```diff
@@ -3,7 +3,7 @@
 import React, {Component} from 'react'
 
 export type Props = {
-  type: 'GPG-export' | 'computer-big' | 'documentation' | 'keybase-update-pause' | 'keybase-update' | 'logo-15' | 'logo-24' | 'logo-128' | 'logo-256' | 'logo-32' | 'logo-64' | 'no-gpg' | 'paper-key' | 'phone-big' | 'phone-oblique' | 'phone-q-r-code-small' | 'phone-q-r-code' | 'phone-text-code-small' | 'phone-text-code' | 'placeholder-avatar' | 'terminal' | 'user-card'
+  type: 'GPG-export' | 'computer-big' | 'documentation' | 'keybase-update-pause' | 'keybase-update' | 'invite-code-m' | 'logo-15' | 'logo-24' | 'logo-128' | 'logo-256' | 'logo-32' | 'logo-64' | 'no-gpg' | 'paper-key' | 'phone-big' | 'phone-oblique' | 'phone-q-r-code-small' | 'phone-q-r-code' | 'phone-text-code-small' | 'phone-text-code' | 'placeholder-avatar' | 'terminal' | 'user-card'
```
Comment (chrisnojima) [inline shared/common-adapters/icon.js.flow]: You could cherry pick my pr. I added all the icons

File: shared/menubar/index.render.desktop.js [hunk 3]
```diff
@@ -409,6 +410,13 @@ const styles = {
     position: 'absolute',
     right: 0,
     top: 5
+  },
+  dz1InputHint: {
+    color: globalColors.black,
+    // This is because normal <Input/> does center-aligned hints by using
+    // width: 100% on position: absolute
+    width: '',
```
Comment (MarcoPolo) [inline shared/menubar/index.render.desktop.js]: What is this suppose to do? Does this reset to initial value?

File: shared/styles/style-guide.desktop.js (hunk 1 above)
Comment (MarcoPolo) [inline shared/styles/style-guide.desktop.js]: This is what `setContentSize` should account for. Let me look why that isn't working. We shouldn't have to account for the menubar ourselves.

Comment (chrisnojima) [inline shared/styles/style-guide.desktop.js]: I'd  be very surprised that this has to be touched. Maybe signup and login
and different sizes in zeplin?

Comment (MarcoPolo) [inline shared/styles/style-guide.desktop.js]: I think the reason he changed it was to account for the top window bar thing with the close/minimize icons. but `setContentSize` should account for that without us having to add the 25px.

Comment (cjb) [inline shared/styles/style-guide.desktop.js]: Signup and login are both 580px in Zeplin, but if I leave it at 580 in globalResizing then I only get ~560px of content, on both OSX and Linux.

File: shared/menubar/index.render.desktop.js (hunk 3 above)
Comment (cjb) [inline shared/menubar/index.render.desktop.js]: Yeah, I'm trying to just remove the width: '100%' in this old DZ1 case.

File: shared/common-adapters/icon.js.flow (hunk 2 above)
Comment (cjb) [inline shared/common-adapters/icon.js.flow]: That's your huge GPG PR, though -- the icons don't even have their own commit inside it -- so I don't see how to include that without messing up the ability to get this diff against master reviewed.

I did use the same icon name (invite-code-m) as you, so the files themselves should merge okay, just leaving a conflict on the Props type.

File: shared/menubar/index.render.desktop.js (hunk 3 above)
Comment (MarcoPolo) [inline shared/menubar/index.render.desktop.js]: Does that actually work? what happens if you use the default value 'auto' (from [here](https://developer.mozilla.org/en-US/docs/Web/CSS/width))

**Comment by MarcoPolo** [comment]
:+1:

File: shared/styles/style-guide.desktop.js (hunk 1 above)
Comment (cjb) [inline shared/styles/style-guide.desktop.js]: Marco's added commits to this branch that fix the window height and revert this height change now.  Thanks!

File: shared/menubar/index.render.desktop.js (hunk 3 above)
Comment (cjb) [inline shared/menubar/index.render.desktop.js]: Yeah, it works.  I've changed it to "width: 'auto'" now, that works too and makes it clearer, thanks.

File: shared/common-adapters/icon.js.flow (hunk 2 above)
Comment (chrisnojima) [inline shared/common-adapters/icon.js.flow]: k my pr went in :)

File: shared/login/signup/invite-code.render.desktop.js [hunk 4]
```diff
@@ -3,36 +3,68 @@
...
+    }
+  }
+
   render () {
-    let inviteRef = null
     const submitInviteCode = () => {
-      const inviteCode = inviteRef && inviteRef.getValue() || ''
-      this.props.onInviteCodeSubmit(inviteCode)
+      this.props.onInviteCodeSubmit(this.state.inviteCode)
+      this.setState({inviteCode: ''})
```
Comment (chrisnojima) [inline shared/login/signup/invite-code.render.desktop.js]: why do this?

File: shared/login/signup/invite-code.render.desktop.js [hunk 5]
```diff
@@ -3,36 +3,68 @@
...
-        <Input ref={r => (inviteRef = r)} hintText='Invite Code' value={this.props.inviteCode || ''} errorText={this.props.inviteCodeErrorText} onEnterKeyDown={submitInviteCode}/>
-        <Button type='Secondary' label='Check' onClick={submitInviteCode}/>
-      </div>
+      <Container onBack={this.props.onBack} style={styles.container}>
+        <Text style={styles.header} type='Header'>Type in your invite code:</Text>
+        <Icon style={styles.icon} type='invite-code-m'/>
+        <Input style={styles.input} hintText='goddess brown result reject' value={this.state.inviteCode || ''} errorText={this.props.inviteCodeErrorText} onEnterKeyDown={submitInviteCode} onChange={event => this.setState({inviteCode: event.target.value})}/>
+        <Button style={styles.button} type='Primary' label='Continue' onClick={submitInviteCode} disabled={!this.state.inviteCode}/>
+        <Text style={styles.text} type='Body'>Not invited?</Text>
+        <Text type='BodyPrimaryLink' onClick={() => {}}>Request an invite code</Text>
```
Comment (chrisnojima) [inline shared/login/signup/invite-code.render.desktop.js]: we shouldn't have this null handler here

File: shared/login/signup/invite-code.render.desktop.js (hunk 4 above)
Comment (cjb) [inline shared/login/signup/invite-code.render.desktop.js]: Thanks, removed.  The point was to disable the submit button after submitting an invalid signup code until you make a change to the text field (`disabled={!this.state.inviteCode}`), but I don't like it after all.

File: shared/login/signup/invite-code.render.desktop.js (hunk 5 above)
Comment (cjb) [inline shared/login/signup/invite-code.render.desktop.js]: Fixed.

**Comment by chrisnojima** [comment]
:+1:

**Pull request merged**